Agents/DataBridgeAgent/README.md

```diff
@@ -0,0 +1,121 @@
+     - Please ensure the version numbers are targeted at the right image. If you are building it, please update the version number accordingly.
+  3) Modify the absolute path of the agent's `config` folder to your absolute file path
+     - For Windows users using WSL on Docker, the file path should start with `/mnt/c/`, which is equivalent to `C://`
+  4) Include this agent service into the stack configuration file at `stack-manager/inputs/config/<STACK-NAME>.json`
+     - Read more in the [Stack Configuration](https://github.com/cambridge-cares/TheWorldAvatar/tree/main/Deploy/stacks/dynamic/stack-manager) section
+  5) Start the stack as per normal
+
+If the agent is successfully started, the endpoint at `http://localhost:3838/data-bridge-agent/status` should return the following message.
+```
+{"Result":"Agent is ready to receive requests."}
+```
+
+### 3. Endpoint
+When successfully built, the agent will be running at `port 3055`. The base url will be `http://localhost:3055/data-bridge-agent`.
+There are currently three routes available:
+
+1. `<base>/status` route:
+   - Returns the current status of the agent through an HTTP `GET` request.
+2. `<base>/sparql` route:
+    - Execute the agent's task through an HTTP `GET` request. This route will transfer data between the specified source and target endpoints.
+    - Before sending the request, please read the instructions.
+    - When transferring triples across non-stack endpoint, please update the source and target SPARQL endpoint in the `<root>/config/endpoint.properties` and send the simple `GET` request.
+      - For non-authenticated endpoints, please leave the corresponding `sparql.<src/target>.user` and `sparql.<src/target>.password` BLANK. 
+      - If either endpoints are secured with username or password, please update their username and password accordingly for that specified endpoint.
+    - When transferring triples from or to the same stack's endpoint, please send the `GET` request with the following parameters:
+      - The `namespace` parameter refers to the stack SPARQL namespace, which is by default `kb`.
+      - The `transfer` parameter indicates whether you wish to transfer triples into or outside the stack. `transfer=in` is for transferring triples from non-stack endpoints into the stack. `transfer=out` is for transferring triples from the stack to non-stack endpoints.
+      - Please ensure that the non-stack endpoint have been updated in the `<root>/config/endpoint.properties` at the right position. The source endpoint must be populated for `transfer=in`, whereas target endpoint must be populated for `transfer=out`. 
+```
+# For any non-stack endpoints
+curl -X GET localhost:3055/data-bridge-agent/sparql
+# For namespaces within the stack
+curl -X GET 'localhost:3838/data-bridge-agent/sparql?namespace=kb&transfer=in'
```

Comment:
As discussed, this raises a couple of points:
1) What is the `transfer` flag being used for? It shouldn't matter whether in or out. The endpoints should just be URLs, and it shouldn't matter whether they are in a stack or not. Then one could also transfer between stacks or within a single stack.
2) Related to that, using the namespace explicitly is not advisable, as this further entrenches our implicit blazegraph dependence. Endpoints should just be URLs - then the namespace (or repository name!) will just be a part of that but have no further significance.

Reply:
Removed the transfer flag for both routes.

The Sparql route now takes in a source and target endpoint parameter as a url. I have also added instructions for its format.

I have refactor the SQL parameters, specifically to handle transfer of data when involving the stack.